src/sidebar/app-controller.js

```diff
@@ -98,6 +100,11 @@ module.exports = function AppController(
 
   // Start the login flow. This will present the user with the login dialog.
   $scope.login = function () {
+    if (serviceConfig(settings)) {
```

Comment:
Or we just expect the site to provide a _different_ `onLogin` function for each service in the array? 

I suppose there's no harm in us sending a unique ID for a service as an arg to `onLogin` just so that the publisher site can identify the service if they do use a single `onLogin` callback for multiple services. 

But then we need a way of uniquely identifying a service.

What uniquely identifies a service?

If `authority` is the unique ID for a service then maybe `services` should it be an object keyed by authority rather than an array? (Because with an array the publisher site can provide us with two services with the same authority.)